### PR TITLE
VIH-6681 introduce global error handler to capture uncaught errors

### PR DESCRIPTION
**Pull request by shaed-parkar**
### JIRA link (if applicable) ###

VIH-6681

### Change description ###

* uncaught errors are logged by the handler
* set appinsights user id to username

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by rehmanab — DISMISSED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/tasks-table/tasks-table.component.ts [hunk 1]
```diff
@@ -54,12 +55,13 @@ export class TasksTableComponent implements OnInit, OnDestroy {
...
     async completeTask(task: TaskResponse) {
         try {
+            this.logger.debug('[TasksTable] - Attempting to complete task', { conference: this.conference.id, task: task.id });
             const updatedTask = await this.vhoQueryService.completeTask(this.conference.id, task.id);
             this.updateTask(updatedTask);
             const payload = new TaskCompleted(this.conference.id, task.id);
             this.eventbus.emit(new EmitEvent(VHEventType.TaskCompleted, payload));
         } catch (error) {
-            this.logger.error(`Failed to complete task ${task.id}`, error);
+            this.logger.error(`[TasksTable] - Failed to complete task ${task.id}`, error);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/tasks-table/tasks-table.component.ts:64]: { conference: this.conference.id, task: task.id }

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/unread-messages-shared/unread-message-base.component.ts [hunk 2]
```diff
@@ -23,14 +23,14 @@ export abstract class UnreadMessagesComponentBase {
     setupSubscribers() {
         this.messagesSubscription$.add(
             this.eventsService.getAdminAnsweredChat().subscribe(message => {
-                this.logger.debug(`an admin has answered ${message.conferenceId}`);
+                this.logger.debug(`[UnreadMesssage] - An admin has answered ${message.conferenceId}`);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/unread-messages-shared/unread-message-base.component.ts:26]: property needed? (x2)

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/pending-tasks/pending-tasks.component.ts [hunk 3]
```diff
@@ -21,10 +21,11 @@ export class PendingTasksComponent implements OnInit, OnDestroy {
 
     ngOnInit() {
         this.setupSubscribers();
+        this.logger.debug('[PendingTasks] - Getting tasks for conference', { conference: this.conferenceId });
         this.queryService
             .getTasksForConference(this.conferenceId)
             .then(tasks => (this.tasks = tasks))
-            .catch(err => this.logger.error(`Failed to get tasks for ${this.conferenceId}`, err));
+            .catch(err => this.logger.error(`[PendingTasks] - Failed to get tasks for ${this.conferenceId}`, err));
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/pending-tasks/pending-tasks.component.ts:28]: property needed?

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/command-centre/command-centre.component.ts [hunk 4]
```diff
@@ -299,7 +298,9 @@ export class CommandCentreComponent implements OnInit, OnDestroy {
             return venue.courtsRooms.some(room => room.selected && participant.lastName === room.courtRoom);
         } else {
             // if the venue could not be found (the venue name is not match the judge first name) will not hide the hearing
-            this.logger.warn(`Venue for judge first name: ${participant.firstName} could not be found in court rooms accounts`);
+            this.logger.warn(
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/command-centre/command-centre.component.ts:301]: property needed? (participant)

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/participant-status/participant-status.component.ts [hunk 5]
```diff
@@ -56,7 +56,10 @@ export class ParticipantStatusComponent implements OnInit {
         try {
             return await this.videoWebService.getParticipantsWithContactDetailsByConferenceId(conferenceId);
         } catch (error) {
-            this.logger.error('There was an error getting the VH Officer dashboard participant status list of names', error);
+            this.logger.error(
+                '[ParticipantStatus] - There was an error getting the VH Officer dashboard participant status list of names',
+                error
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/participant-status/participant-status.component.ts:61]: property conferenceId

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/command-centre/command-centre.component.ts [hunk 6]
```diff
@@ -134,7 +133,7 @@ export class CommandCentreComponent implements OnInit, OnDestroy {
     }
 
     onConferenceSelected(conference: ConferenceForVhOfficerResponse) {
-        this.logger.info(`Conference ${conference.id} selected`);
+        this.logger.info(`[CommandCentre] - Conference ${conference.id} selected`);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/command-centre/command-centre.component.ts:136]: prop

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/tasks-table/tasks-table.component.ts [hunk 7]
```diff
@@ -27,11 +27,12 @@ export class TasksTableComponent implements OnInit, OnDestroy {
         this.retrieveConference(this.conferenceId)
             .then(async conference => {
                 this.conference = conference;
+                this.logger.debug('[TasksTable] - Getting tasks for conference', { conference: this.conferenceId });
                 this.tasks = await this.retrieveTasksForConference(this.conference.id);
                 this.loading = false;
             })
             .catch(err => {
-                this.logger.error(`Failed to init tasks list for conference ${this.conferenceId}`, err);
+                this.logger.error(`[TasksTable] - Failed to init tasks list for conference ${this.conferenceId}`, err);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/tasks-table/tasks-table.component.ts:35]: this.conference.id

**Review by levibutler — COMMENTED** [review]
nice! ive made a couple comments, nothing much just around the new properties. Feel free to ignore

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/video-call.service.ts [hunk 8]
```diff
@@ -188,6 +188,7 @@ export class VideoCallService {
     }
 
     updatePreferredLayout(conferenceId: string, layout: HearingLayout) {
+        this.logger.info(`[VideoCallService] - Updating preferred layout`, { conferenceId, layout });
```
[the remark this answers is not in the document]
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/video-call.service.ts:191]: no you're right

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/unread-messages/unread-messages.component.ts [hunk 9]
```diff
@@ -30,10 +30,11 @@ export class UnreadMessagesComponent extends UnreadMessagesComponentBase impleme
 
     ngOnInit() {
         this.setupSubscribers();
+        this.logger.debug('[UnreadMessages] - Getting unread message count for conference', { conference: this.hearing.id });
         this.videoWebService
             .getUnreadMessageCountForConference(this.hearing.id)
             .then(response => (this.unreadMessages = response.number_of_unread_messages_conference))
-            .catch(err => this.logger.error(`Failed to get unread vho messages for ${this.hearing.id}`, err));
+            .catch(err => this.logger.error(`[UnreadMessages] - Failed to get unread vho messages for ${this.hearing.id}`, err));
```
[the remark this answers is not in the document]
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/unread-messages/unread-messages.component.ts:37]: Hearing is the ng model that takes the raw dto from the api and has additional functionality

**Review by npoxon — APPROVED** [review]

**Pull request merged**